src/hotspot/cpu/x86/x86.ad

```diff
@@ -7287,18 +7292,31 @@ instruct castFtoI_reg_avx(vec dst, vec src, vec xtmp1, vec xtmp2, vec xtmp3, vec
   ins_pipe( pipe_slow );
 %}
 
-instruct castFtoI_reg_evex(vec dst, vec src, vec xtmp1, vec xtmp2, kReg ktmp1, kReg ktmp2, rRegP scratch, rFlagsReg cr) %{
-  predicate((VM_Version::supports_avx512vl() ||
-             Matcher::vector_length_in_bytes(n) == 64) &&
-             Matcher::vector_element_basic_type(n) == T_INT);
+instruct castFtoX_reg_evex(vec dst, vec src, vec xtmp1, vec xtmp2, kReg ktmp1, kReg ktmp2, rRegP scratch, rFlagsReg cr) %{
+  predicate(((VM_Version::supports_avx512vl() ||
+              Matcher::vector_length_in_bytes(n) == 64)) &&
+             is_integral_type(Matcher::vector_element_basic_type(n)));
```

Comment:
Do we need some of these conditions since you have them already in `match_rule_supported_vector()`?

Reply:
Yes it was needed to select between the rules. On platforms that don't support avx512vl, we use AVX512 instructions only for 512-bit vectors and AVX instructions for < 64 byte vectors.